sql/src/test/java/io/crate/operation/scalar/ArrayCatFunctionTest.java

```diff
@@ -0,0 +1,226 @@
+                Literal.NULL);
+    }
+
+    @Test
+    public void testNullArgumentsWithoutCheckingForArrays() throws Exception {
+        ArrayCatFunction function = getFunction(arrayOfIntegerType, arrayOfIntegerType);
+
+        Input[] inputs = new Input[]{
+                Literal.NULL,
+                null
+        };
+
+        Object[] expected = new Object[] {};
+        Object[] evaluate = function.evaluate(inputs);
+        assertThat(evaluate, is(expected));
+    }
+
+    @Test
+    public void testZeroArguments() throws Exception {
+        expectedException.expect(ElasticsearchIllegalArgumentException.class);
+        expectedException.expectMessage("array_cat function requires 2 arguments");
+        assertEval(null);
+    }
+
+    @Test
+    public void testOneArgument() throws Exception {
+        expectedException.expect(ElasticsearchIllegalArgumentException.class);
+        expectedException.expectMessage("array_cat function requires 2 arguments");
+        assertEval(new Object[]{}, Literal.newLiteral(new Object[]{1}, new ArrayType(DataTypes.INTEGER)));
+    }
+
+    @Test
+    public void testThreeArguments() throws Exception {
+        expectedException.expect(ElasticsearchIllegalArgumentException.class);
+        expectedException.expectMessage("array_cat function requires 2 arguments");
+        assertEval(new Object[]{},
+                Literal.newLiteral(new Object[]{1}, new ArrayType(DataTypes.INTEGER)),
+                Literal.newLiteral(new Object[]{2}, new ArrayType(DataTypes.INTEGER)),
+                Literal.newLiteral(new Object[]{3}, new ArrayType(DataTypes.INTEGER)));
+    }
+
+    @Test
+    public void testDifferentConvertableInnerTypes() throws Exception {
+        assertEval(
+                new Object[]{1, 1},
+                Literal.newLiteral(new Object[]{1},  arrayOfIntegerType),
+                Literal.newLiteral(new Object[]{1L}, arrayOfLongType));
+    }
+
+    @Test
+    public void testStringToNumberCast() throws Exception {
+        assertEval(
+                new Object[]{1, 2},
+                Literal.newLiteral(new Object[]{1},      arrayOfIntegerType),
+                Literal.newLiteral(new Object[]{"2"},    arrayOfStringType));
+    }
+
+    @Test
+    public void testNumberToStringCast() throws Exception {
+        assertEval(
+                new Object[]{new BytesRef("2"), new BytesRef("1")},
+                Literal.newLiteral(new Object[]{"2"},    arrayOfStringType),
+                Literal.newLiteral(new Object[]{1},      arrayOfIntegerType));
+    }
+
+    @Test
+    public void testDifferentUnconvertableInnerTypes() throws Exception {
+        expectedException.expect(NumberFormatException.class);
+        assertEval(
+                null,
+                Literal.newLiteral(new Object[]{1},              arrayOfIntegerType),
+                Literal.newLiteral(new Object[]{"foo","bar"},    arrayOfStringType));
+    }
```

Comment:
pls also test eval of 2 array types whose inner types are not convertable like Ip and Boolean